scripts/merge-changelogs.ts

```diff
@@ -0,0 +1,191 @@
+    if (!match.groups?.commit && !match.groups?.depsCommit) {
+      throw new Error(
+        `No commit hash found for change set in ${packageName} for version ${version}!`
+      );
+    }
+    if (
+      !validMessageTypes.includes(match.groups?.type as any) &&
+      !match[0].includes('Updated dependencies')
+    ) {
+      throw new Error(
+        `Change set has an incorrect or missing type for change set in ${packageName} for version ${version}!`
+      );
+    }
+    const message = match.groups?.message?.trim() || match.groups?.deps?.trim();
+    if (message) {
+      results.push({
+        packageNames: [packageName],
+        version,
+        commit: match.groups?.commit || match.groups?.depsCommit,
+        type: (match.groups?.type as any) || 'Updated Dependencies',
+        message: match.groups?.message?.trim() || match.groups?.deps?.trim()
+      });
+    }
+  }
+  return results;
+}
+
+function parseChangelog(changelog: string): Message[] {
+  const packageName = getPackageName(changelog);
+  const versions = splitByVersion(changelog);
+  return versions
+    .map(({ version, content }) => parseContent(content, version, packageName))
+    .flat();
+}
+
+function writeMessagesOfType(
+  messages: Message[],
+  type: typeof validMessageTypes[number]
+): string {
+  return messages.some(msg => msg.type === type)
+    ? `## ${type}
+
+${messages
+  .filter(msg => msg.type === type)
+  .map(
+    msg => `- [${msg.packageNames.join(', ')}] ${msg.message} (${msg.commit})`
+  )
+  .join('\n')}`
+    : '';
+}
+
+function mergeMessages(parsedMessages: Message[]): Message[] {
+  return parsedMessages.reduce((prev, curr) => {
+    const sameMessage = prev.find(
+      msg =>
+        msg.message === curr.message &&
+        msg.version === curr.version &&
+        msg.type === curr.type
+    );
+    if (sameMessage) {
+      sameMessage.packageNames.push(curr.packageNames[0]);
+      return prev;
+    }
+    return [...prev, curr];
+  }, [] as Message[]);
+}
+
+function createNewSection(version: string, messages: Message[]): string {
+  return `
+# ${version}
+
+Release Date: TBD<br>
+API Docs: https://sap.github.io/cloud-sdk/api/${version}<br>
+Blog: TBD<br>
+
+${writeMessagesOfType(messages, 'Compatibility Note')}
+
+${writeMessagesOfType(messages, 'New Functionality')}
+
+${writeMessagesOfType(messages, 'Improvement')}
+
+${writeMessagesOfType(messages, 'Fixed Issue')}
+
+${
+  messages.filter(msg => msg.type === 'Updated Dependencies')
+    ? '## Updated Dependencies\n'
+    : ''
+}
+${messages
+  .filter(msg => msg.type === 'Updated Dependencies')
+  .map(
+    msg =>
+      `- [${msg.packageNames.join(', ')}] Updated Dependencies (${
+        msg.commit
+      })\n  ${msg.message}`
+  )
+  .join('\n')}
+`;
+}
+
+async function formatChangelog(parsedChangelog: Message[]): Promise<string> {
+  let unifiedChangelog = await readFile('CHANGELOG.md', { encoding: 'utf8' });
+  const relevantMessages = parsedChangelog.filter(
```

Comment:
Now I am confused and have to look at the code. I saw that the CHANGELOG.md in the sub-packages are removed and I thought they are not needed.

Reply:
ah now I got it. So the changelog files in the packages will stay in the new format introduced by changeset and will accumulate the changes over time.

I would keep the per package changeset files since they could be handy if you want to check changes per package.

Reply:
[pp] Perhas find a better name `relevantMessages` -> `newMessagesWhichAreNotInUnifiedChangelog` or something shorter but relevant is a bit vague.